pkg/agent/resourcemanager/fetcher/kubelet/topology/topology_adapter.go

```diff
@@ -488,6 +502,50 @@ func (p *topologyAdapterImpl) getZoneAllocations(podList []*v1.Pod, podResources
 	return zoneAllocationsMap, nil
 }
 
+// revisePodAllocated is to revise pod allocated according to its qos level
+func (p *topologyAdapterImpl) revisePodAllocated(pod *v1.Pod, podAllocated map[util.ZoneNode]*v1.ResourceList) error {
+	qosLevel, err := p.qosConf.GetQoSLevel(pod, map[string]string{})
+	if err != nil {
+		return err
+	}
+
+	switch qosLevel {
+	case apiconsts.PodAnnotationQoSLevelSharedCores:
+		// revise shared_cores pod allocated according to its numa binding
+		return p.reviseSharedCoresPodAllocated(pod, podAllocated)
+	default:
+		return nil
+	}
+}
+
+// reviseSharedCoresPodAllocated is to revise shared_cores pod allocated according to its numa binding
+func (p *topologyAdapterImpl) reviseSharedCoresPodAllocated(pod *v1.Pod, podAllocated map[util.ZoneNode]*v1.ResourceList) error {
+	if !qos.IsPodNumaBinding(p.qosConf, pod) {
+		return nil
+	}
+
+	var bindingNumaNode *util.ZoneNode
+	for zoneNode, resourceList := range podAllocated {
```

Comment:
move this judgment to util since we may need to judge whether a pod is only for single-numa multiple times during execution